packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -465,7 +467,9 @@ export function appendChild(
   parentInstance: Instance,
   child: Instance | TextInstance,
 ): void {
-  parentInstance.appendChild(child);
+  if (parentInstance.tagName === 'TEMPLATE')
+    (parentInstance: HTMLTemplateElement).content.appendChild(child);
```

Comment:
I haven't looked at the issue / feature request yet, but I want to flag that this is pretty concerning because it's one of the hottest paths in the entire codebase, and introducing even a small check in it can have significant perf implications.

Reply:
I understand your concerns and it was one of my concerns too. I'm open to moving this check wherever possible; I just haven't figured out where would be a better place to put this.

Reply:
Perhaps we could treat them like a portal when creating a template element... I think I'll try this way instead but please let me know if you have any other ideas!